nixos/modules/services/wayland/cage.nix

```diff
@@ -0,0 +1,100 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cage;
+in {
+  options.services.cage.enable = mkEnableOption "cage kiosk service";
+
+  options.services.cage.user = mkOption {
+    type = types.str;
+    default = "demo";
+    description = ''
+      User to log-in as.
+    '';
+  };
+
+  options.services.cage.extraArguments = mkOption {
+    type = types.listOf types.str;
+    default = [];
+    defaultText = "[]";
+    description = "Additional command line arguments to pass to Cage.";
+    example = ["-d"];
+  };
+
+  options.services.cage.program = mkOption {
+    type = types.path;
+    default = "${pkgs.xterm}/bin/xterm";
+    description = ''
+      Program to run in cage.
+    '';
+  };
+
+  config = mkIf cfg.enable {
+
+    # The service is partially based off of the one provided in the
+    # cage wiki at
+    # https://github.com/Hjdskes/cage/wiki/Starting-Cage-on-boot-with-systemd.
+    systemd.services."cage-tty1" = {
+      enable = true;
+      after = [
+        "systemd-user-sessions.service"
+        "plymouth-start.service"
+        "plymouth-quit.service"
+        "dbus.socket"
```

Comment:
Where is socket activated dbus declared?

Reply:
It's part of the dbus package:

```
 flokli@xps  ~  systemctl status dbus.socket                                                       ✔  10002  20:04:51 
● dbus.socket - D-Bus System Message Bus Socket
   Loaded: loaded (/nix/store/4mq1yhlsx59p20a1wklm4smp20cqps3d-dbus-1.12.16/etc/systemd/system/dbus.socket; linked; vendor >
   Active: active (running) since Fri 2020-02-28 13:04:27 PST; 7h ago
   Listen: /run/dbus/system_bus_socket (Stream)
   CGroup: /system.slice/dbus.socket

Feb 28 13:04:27 xps.flokli.de systemd[1]: Listening on D-Bus System Message Bus Socket.
```

Reply:
Oh, I was thinking of services.dbus.socketActivated

Reply:
https://github.com/Hjdskes/cage/wiki/Starting-Cage-on-boot-with-systemd mentions

> # D-Bus is necessary for contacting logind, which is required.

However, `systemd-logind.service` itself pulls in `dbus.socket` since https://github.com/systemd/systemd/commit/8f9c6fe5ff1d59001aecbf3fbf9ca0ed7ff28ba7, so we can drop it.

I did this, and the test still succeeds.